sdk/core/core-http/src/policies/compressPolicy.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BaseRequestPolicy, RequestPolicy, RequestPolicyOptions } from './requestPolicy';
+import { WebResource } from '../webResource';
+import { HttpOperationResponse } from '../httpOperationResponse';
+
+/**
+ * Options for how HTTP connections should handle the gzip/defalte encoding for future
+ * requests.
+ */
+export interface CompressOptions {
+  /*
+   * When true, enables the support of gzip/defalte content encoding in node-fetch client
+   * Defaults to true.
+   */
+  enable: boolean;
+}
+
+export const DefaultCompressOptions: CompressOptions = {
+  enable: true
+}
+
+export function compressPolicy(compressOptions?: CompressOptions) {
+  return {
+    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+      return new CompressPolicy(nextPolicy, options, compressOptions || DefaultCompressOptions);
```

Comment:
instead of `compressOptions || DefaultCompressOptions` here, can we just make it the default value of `compressOptions` in `compressPolicy`'s signature?

Reply:
Good point! Fixed.